pkgs/by-name/pi/pioasm/package.nix

```diff
@@ -0,0 +1,37 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, ninja
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "pioasm";
+  version = "1.5.1";
+
+  src = fetchFromGitHub {
+    owner = "raspberrypi";
+    repo = "pico-sdk";
+    rev = finalAttrs.version;
+    hash = "sha256-JNcxd86XNNiPkvipVFR3X255boMmq+YcuJXUP4JwInU=";
+  };
+  sourceRoot = "${finalAttrs.src.name}/tools/pioasm";
+
+  nativeBuildInputs = [ cmake ninja ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D pioasm $out/bin/pioasm
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Assemble PIO programs for Raspberry Pi Pico";
+    homepage = "https://github.com/raspberrypi/pico-sdk";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ emilytrau ];
+    platforms = platforms.unix;
```

Comment:
```suggestion
    platforms = platforms.unix;
    mainProgram = "pioasm";
```

Reply:
This should already be default since it's the same as the package name?

Reply:
Can't remember where it read it, but you can see from the comment on line 149 that it is deprecated.

[https://github.com/NixOS/nixpkgs/blob/master/lib/meta.nix](https://github.com/NixOS/nixpkgs/blob/master/lib/meta.nix)

Let me know if I need to provide more details. (I'm on my phone, hence the short answer)

Reply:
ooo i learned something new thanks!